### ISSUE TITLE
i3status 2.12 fails to build on MacOS

### ISSUE THREAD

**Issue by larryhynes** [i3/i3status#280]
Hi

The latest release of i3status, 2.12, fails to build on MacOS (10.13.4). (Previous release builds fine.)

Is supporting MacOS something that you are interested in? (I hope it is!)

First thing that I notice is that on MacOS `socket.h` seems to be at `/usr/include/sys/socket.h`, and not `/sys/socket.h`. I can help with testing / debugging if you like?

```
$ make
 CC i3status.c
 CC src/auto_detect_format.c
 CC src/first_network_device.c
 CC src/general.c
 CC src/output.c
src/print_battery_info.c:113:11: warning: unused variable 'outwalk' [-Wunused-variable]
    char *outwalk = buffer;
          ^
1 warning generated.
 CC src/print_battery_info.c
src/print_cpu_temperature.c:54:12: warning: unused function 'read_temperature' [-Wunused-function]
static int read_temperature(char *thermal_zone, temperature_t *temperature) {
           ^
1 warning generated.
 CC src/print_cpu_temperature.c
src/print_cpu_usage.c:58:22: warning: unused variable 'curr_all' [-Wunused-variable]
    struct cpu_usage curr_all = {0, 0, 0, 0, 0};
                     ^
src/print_cpu_usage.c:59:20: warning: unused variable 'diff_total' [-Wunused-variable]
    int diff_idle, diff_total, diff_usage;
                   ^
src/print_cpu_usage.c:59:9: warning: unused variable 'diff_idle' [-Wunused-variable]
    int diff_idle, diff_total, diff_usage;
        ^
src/print_cpu_usage.c:45:25: warning: unused variable 'prev_all' [-Wunused-variable]
static struct cpu_usage prev_all = {0, 0, 0, 0, 0};
                        ^
4 warnings generated.
 CC src/print_cpu_usage.c
 CC src/print_ddate.c
 CC src/print_disk_info.c
 CC src/print_eth_info.c
 CC src/print_ip_addr.c
 CC src/print_ipv6_addr.c
 CC src/print_load.c
 CC src/print_path_exists.c
 CC src/print_run_watch.c
 CC src/print_time.c
 CC src/print_volume.c
src/print_wireless_info.c:489:57: error: use of undeclared identifier 'AF_INET'
    char *ipv4_address = sstrdup(get_ip_addr(interface, AF_INET));
                                                        ^
src/print_wireless_info.c:490:57: error: use of undeclared identifier 'AF_INET6'
    char *ipv6_address = sstrdup(get_ip_addr(interface, AF_INET6));
                                                        ^
src/print_wireless_info.c:612:5: warning: implicit declaration of function 'free' is invalid in C99
      [-Wimplicit-function-declaration]
    free(ipv4_address);
    ^
1 warning and 2 errors generated.
make: *** [src/print_wireless_info.o] Error 1
```

**Comment by Airblader** [comment]
> Is supporting MacOS something that you are interested in? (I hope it is!)

If the effort is reasonable I don't see a reason not to make it work there as well.

**Comment by eplanet** [comment]
Seems to only be a matter of finding mandatory `AF_INET` and `AF_INET6` constants declaration somewhere.

About `sys/socket.h` it shouldn't be a problem at all since it's included using `#include <sys/socket.h>`, so the compiler will adapt and find the header correctly. I assume you'd have been stuck on include errors otherwise.

**Comment by larryhynes** [comment]
May of use:

```
$ ag AF_INET /usr/include

/usr/include/apr-1/apr_network_io.h
142:#define APR_INET     AF_INET
156:#define APR_INET6    AF_INET6

/usr/include/cups/http.h
83:#if defined(AF_INET6) && !defined(s6_addr32)
95:#endif /* AF_INET6 && !s6_addr32 */
417:#ifdef AF_INET6
419:#endif /* AF_INET6 */

/usr/include/gssapi/gssapi.h
196:#define GSS_C_AF_INET       2

/usr/include/netinet/in_pcb.h
89: * PCB with AF_INET6 null bind'ed laddr can receive AF_INET input packet.
90: * So, AF_INET6 null laddr is also used as AF_INET null laddr, by utilizing
279:#define	IN6P_IPV6_V6ONLY	0x00008000 /* restrict AF_INET6 socket for v6 */

/usr/include/netinet6/in6.h
172:	sa_family_t	sin6_family;	/* AF_INET6 (sa_family_t) */

/usr/include/sys/socket.h
356:#define	AF_INET		2		/* internetwork: UDP, TCP, etc. */
387:#define	AF_INET6	30		/* IPv6 */
451:#define	PF_INET		AF_INET
474:#define	PF_RTIP		pseudo_AF_RTIP	/* same format as AF_INET */
479:#define	PF_INET6	AF_INET6
```

**Comment by larryhynes** [comment]
Re: pulseaudio, that's a package that can be installed with e.g. homebrew, but I don't think apple includes it natively.

**Comment by eplanet** [comment]
In that case it works correctly with a rather simple fix, proposed in #281.

**Comment by larryhynes** [comment]
Brilliant, thank you. That's building fine here now:

```
$ make
 CC i3status.c
 CC src/auto_detect_format.c
 CC src/first_network_device.c
 CC src/general.c
 CC src/output.c
src/print_battery_info.c:113:11: warning: unused variable 'outwalk' [-Wunused-variable]
    char *outwalk = buffer;
          ^
1 warning generated.
 CC src/print_battery_info.c
src/print_cpu_temperature.c:54:12: warning: unused function 'read_temperature' [-Wunused-function]
static int read_temperature(char *thermal_zone, temperature_t *temperature) {
           ^
1 warning generated.
 CC src/print_cpu_temperature.c
src/print_cpu_usage.c:58:22: warning: unused variable 'curr_all' [-Wunused-variable]
    struct cpu_usage curr_all = {0, 0, 0, 0, 0};
                     ^
src/print_cpu_usage.c:59:9: warning: unused variable 'diff_idle' [-Wunused-variable]
    int diff_idle, diff_total, diff_usage;
        ^
src/print_cpu_usage.c:59:20: warning: unused variable 'diff_total' [-Wunused-variable]
    int diff_idle, diff_total, diff_usage;
                   ^
src/print_cpu_usage.c:45:25: warning: unused variable 'prev_all' [-Wunused-variable]
static struct cpu_usage prev_all = {0, 0, 0, 0, 0};
                        ^
4 warnings generated.
 CC src/print_cpu_usage.c
 CC src/print_ddate.c
 CC src/print_disk_info.c
 CC src/print_eth_info.c
 CC src/print_ip_addr.c
 CC src/print_ipv6_addr.c
 CC src/print_load.c
 CC src/print_path_exists.c
 CC src/print_run_watch.c
 CC src/print_time.c
 CC src/print_volume.c
src/print_wireless_info.c:613:5: warning: implicit declaration of function 'free' is invalid in C99
      [-Wimplicit-function-declaration]
    free(ipv4_address);
    ^
1 warning generated.
 CC src/print_wireless_info.c
 CC src/process_runs.c
 CC src/pulse.c
 LD i3status
```
